semantic_conventions/trace/http.yaml

```
@@ -117,6 +117,10 @@ groups:
         type: int
         brief: >
           The ordinal number of request re-sending attempt.
```

```suggestion
          The ordinal number of request re-sending attempt (for any reason, including redirects).
```

I think the name "retry_count" is not optimal and I would at least call this aspect out in the brief.

---

Hm, should we rename it to `resend_count` then? "Resend" might be a little less intuitive than "retry", but on the other hand it doesn't introduce any new concept - it relies on the HTTP RFC terminology

---

I'm more inclined in retry, resend somehow reads odd to me. My comment around inconsistency was more to always use retry everywhere.

Also, I'm not entirely sure what is the process If we decide to rename the attribute. The conventions are experimental still, but I _think_ we will need to introduce a schema change in that case?

---

> My comment around inconsistency was more to always use retry everywhere

That would be consistent and easy to understand, but it would be wrong 😉

---

Ah got it, now it makes sense and thanks for the background on it. If we don't want to rename the existing, there could be another just for the "non error" cases like redirect but that's probably an overkill, so maybe renaming it's the way to go. Then I guess we will do need a schema change?

---

If we had two counters that would be very confusing and open many questions IMHO, the first obvious one is that of categorizing. Retries and network errors are clear, but what about re-sends after an authentication challenge for example? Error or not?
The second one, do you reset the other counter to zero if you increment the first? No matter what you do, you don't get an unambiguous order of the requests from the two counters anymore. You would need a third counter.
So I would definitely say we should stay with a single counter, no matter what it is called.

---

> Then I guess we will do need a schema change?

Yes, I suppose so.

I'll add a commit renaming this attribute to `resend_count`, let's see how it looks like.